adapters/httpapi_curl.c

```diff
@@ -1014,10 +1068,50 @@ HTTPAPI_RESULT HTTPAPI_CloneOption(const char* optionName, const void* value, co
             }
             else
             {
-                /*return OK when the certificate has been clones successfully*/
+                /*return OK when the certificate has been cloned successfully*/
                 result = HTTPAPI_OK;
             }
         }
+#ifdef USE_OPENSSL
+        else if (strcmp(OPTION_OPENSSL_PRIVATE_KEY_TYPE, optionName) == 0)
+        {
+            const OPTION_OPENSSL_KEY_TYPE type = **(const OPTION_OPENSSL_KEY_TYPE**)value;
+            if (type == KEY_TYPE_DEFAULT || type == KEY_TYPE_ENGINE)
+            {
+                OPTION_OPENSSL_KEY_TYPE* temp = malloc(sizeof(OPTION_OPENSSL_KEY_TYPE));
+                if (temp == NULL)
+                {
+                    LogError("unable to clone x509PrivatekeyType");
+                    result = HTTPAPI_ERROR;
+                }
+                else
+                {
+                    *temp = type;
+                    *savedValue = temp;
+                    result = HTTPAPI_OK;
+                }
+            }
+            else
+            {
+                LogError("Unknown x509PrivatekeyType: %i", type);
+                result = HTTPAPI_ERROR;
+            }
+        }
+        else if (strcmp(OPTION_OPENSSL_ENGINE, optionName) == 0)
+        {
+            /*this is getting the engine. In this case, value is a pointer to a const char* that contains the engine as a null terminated string*/
```

Comment:
Crashing here: the value is a char* not a char**.

Reply:
This value is passed similarly to line 1118 below: 
`if (mallocAndStrcpy_s((char**)savedValue, value) != 0)`

Reply:
Thanks, I missed that. I fixed it now.